### PR TITLE
[C++] Fix issue when overwriting feather files

### LINKED ISSUE #249: Feather IO subsystem behaves improperly when the user attempts to overwrite an existing file
Files are not being truncated to 0 bytes before being written if they already exist -- this will cause corrupted metadata in a lot of cases if the data written is smaller than the original file (because the writes still happen from the beginning of the file).

There are already a lot of bug reports (#248, ...) that are caused by this problem. We should fix and make a bugfix release as soon as practical

### PR DESCRIPTION
**Pull request by guillett**
Fixes #249.

* Added R and Python tests and
* Added Truncate flag in the open call.

It may also impact #232 (as per the latest comment there).

### CODE REVIEW

File: python/feather/tests/test_reader.py [hunk 1]
```diff
@@ -287,6 +287,18 @@ def test_read_columns(self):
...
+        path = random_path()
+
+        num_values = 100
+        np.random.seed(0)
+
+        values = np.random.randint(0, 10, size=num_values)
+        feather.write_dataframe(pd.DataFrame({'ints':values}), path)
+
+        df = pd.DataFrame({'ints':values[0:num_values//2]})
+        self._check_pandas_roundtrip(df, path=path)
```
Comment (wesm) [inline python/feather/tests/test_reader.py]: Does this test fail without the fix?

Comment (guillett) [inline python/feather/tests/test_reader.py]: Yes it does fail without the fix cf. just-created build with tests only https://travis-ci.org/guillett/feather/builds/170437697.

**Comment by wesm** [comment]
Cool, I also reproduced the segfault locally

**Comment by wesm** [comment]
I cancelled the OS X build because Travis has been really clogged up lately. Merging this

**Pull request closed without merging**